proj/cmake/platform_msw.cmake

```diff
@@ -3,55 +3,102 @@ cmake_minimum_required( VERSION 2.8 FATAL_ERROR )
+    ${CINDER_SRC_DIR}/cinder/msw/StackWalker.cpp
+    ${CINDER_SRC_DIR}/cinder/ImageSourceFileWic.cpp
+    ${CINDER_SRC_DIR}/cinder/ImageTargetFileWic.cpp
+    ${CINDER_SRC_DIR}/cinder/UrlImplWinInet.cpp
+    ${CINDER_SRC_DIR}/glload/wgl_load_cpp.cpp
+    ${CINDER_SRC_DIR}/glload/wgl_load.c
+    ${CINDER_SRC_DIR}/AntTweakBar/TwDirect3D11.cpp
 )
 
 list( APPEND SRC_SET_APP_MSW
-	# TODO: should these two files be added to "cinder\\app" group?
-	${CINDER_SRC_DIR}/cinder/app/AppScreenSaver.cpp
-	#${CINDER_SRC_DIR}/cinder/app/RendererDx.cpp
-
-	${CINDER_SRC_DIR}/cinder/app/msw/AppImplMsw.cpp
-	${CINDER_SRC_DIR}/cinder/app/msw/AppImplMswBasic.cpp
-	${CINDER_SRC_DIR}/cinder/app/msw/AppImplMswScreenSaver.cpp
-	${CINDER_SRC_DIR}/cinder/app/msw/AppMsw.cpp
-	${CINDER_SRC_DIR}/cinder/app/msw/PlatformMsw.cpp
-	${CINDER_SRC_DIR}/cinder/app/msw/RendererImpl2dGdi.cpp
-	#${CINDER_SRC_DIR}/cinder/app/msw/RendererImplDx.cpp
-	#${CINDER_SRC_DIR}/cinder/app/msw/RendererImplGlAngle.cpp
-	${CINDER_SRC_DIR}/cinder/app/msw/RendererImplGlMsw.cpp
+    # TODO: should these two files be added to "cinder\\app" group?
+    ${CINDER_SRC_DIR}/cinder/app/AppScreenSaver.cpp
+    #${CINDER_SRC_DIR}/cinder/app/RendererDx.cpp
+
+    ${CINDER_SRC_DIR}/cinder/app/msw/AppImplMsw.cpp
+    ${CINDER_SRC_DIR}/cinder/app/msw/AppImplMswBasic.cpp
+    ${CINDER_SRC_DIR}/cinder/app/msw/AppImplMswScreenSaver.cpp
+    ${CINDER_SRC_DIR}/cinder/app/msw/AppMsw.cpp
+    ${CINDER_SRC_DIR}/cinder/app/msw/PlatformMsw.cpp
+    ${CINDER_SRC_DIR}/cinder/app/msw/RendererImpl2dGdi.cpp
+    #${CINDER_SRC_DIR}/cinder/app/msw/RendererImplDx.cpp
+    #${CINDER_SRC_DIR}/cinder/app/msw/RendererImplGlAngle.cpp
+    ${CINDER_SRC_DIR}/cinder/app/msw/RendererImplGlMsw.cpp
 )
 
 list( APPEND SRC_SET_AUDIO_MSW
-	${CINDER_SRC_DIR}/cinder/audio/msw/ContextWasapi.cpp
-	${CINDER_SRC_DIR}/cinder/audio/msw/DeviceManagerWasapi.cpp
-	${CINDER_SRC_DIR}/cinder/audio/msw/FileMediaFoundation.cpp
-	${CINDER_SRC_DIR}/cinder/audio/msw/MswUtil.cpp
+    ${CINDER_SRC_DIR}/cinder/audio/msw/ContextWasapi.cpp
+    ${CINDER_SRC_DIR}/cinder/audio/msw/DeviceManagerWasapi.cpp
+    ${CINDER_SRC_DIR}/cinder/audio/msw/FileMediaFoundation.cpp
+    ${CINDER_SRC_DIR}/cinder/audio/msw/MswUtil.cpp
 )
 
 list( APPEND SRC_SET_CINDER_AUDIO_DSP
-	${CINDER_SRC_DIR}/cinder/audio/dsp/ooura/fftsg.cpp
-	${CINDER_SRC_DIR}/cinder/audio/dsp/ConverterR8brain.cpp
+    ${CINDER_SRC_DIR}/cinder/audio/dsp/ooura/fftsg.cpp
+    ${CINDER_SRC_DIR}/cinder/audio/dsp/ConverterR8brain.cpp
 )
 
 list( APPEND CINDER_SRC_FILES
-	${SRC_SET_MSW}
-	${SRC_SET_APP_MSW}
-	${SRC_SET_AUDIO_MSW}
-	${SRC_SET_CINDER_AUDIO_DSP}
+    ${SRC_SET_MSW}
+    ${SRC_SET_APP_MSW}
+    ${SRC_SET_AUDIO_MSW}
+    ${SRC_SET_CINDER_AUDIO_DSP}
 )
 
-source_group( "cinder\\msw"       	FILES ${SRC_SET_MSW} )
-source_group( "cinder\\app\\msw"  	FILES ${SRC_SET_APP_MSW} )
+# Option for using GStreamer under MSW.
+if( CINDER_MSW )
+    option( CINDER_MSW_USE_GSTREAMER "Use GStreamer for video playback." OFF )
```

Comment:
I was thinking this would be a nice global option, for all platforms even. By default it can be on for linux, off for mac and windows. Not sure how we'd handle who gets priority for the different video backends, though (but then, qtime on windows is on it's last leg anyhow).

Reply:
Yes, that would make sense probably. Lets see how everything pans out after a bit of testing and if people are happy with everything we can make this a global `CINDER_USE_GSTREAMER` option. In terms of priority I guess we can keep GStreamer default for Linux, second option for macOS and on Windows we would still have to see what makes more sense I guess..